modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapModelLoaderDelegate.java

```diff
@@ -51,16 +52,16 @@
   public SoapModelLoaderDelegate(Class<?> extensionType, String version) {
     this.extensionType = extensionType;
     this.version = version;
-    this.typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
-    this.serviceProviderDeclarer = new SoapServiceProviderDeclarer(typeLoader);
+    this.typeLoader = new DefaultExtensionsTypeLoaderFactory().createTypeLoader(extensionType.getClassLoader());
```

Comment:
No, this flavor doesn't use the ExtensionFieldHandler making object introspection work differently.
This didn't happen before because the TypeLoader was injected every time that metadatatypes were required